dex/networks/eth/tokens.go

```diff
@@ -0,0 +1,145 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+//go:build lgpl
+// +build lgpl
+
+package eth
+
+import (
+	"fmt"
+	"math/big"
+
+	"decred.org/dcrdex/dex"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Token is the definition of an ERC20 token, including all of its network and
+// version variants.
+type Token struct {
+	*dex.Token
+	// NetTokens is a mapping of token addresses for each network available.
+	NetTokens map[dex.Network]*NetToken `json:"netAddrs"`
+	// EVMFactor allows for arbitrary ERC20 decimals. For an ERC20 contract,
+	// the relation
+	//    math.Log10(UnitInfo.Conventional.ConversionFactor) + Token.EVMFactor = decimals
+	// should hold true.
+	// Since most assets will use a value of 9 here, a default value of 9 will
+	// be used in AtomicToEVM and EVMToAtomic if EVMFactor is not set.
+	EVMFactor *int64 `json:"evmFactor"` // default 9
+}
+
+// factor calculates the conversion factor to and from DEX atomic units to the
+// units used for EVM operations.
+func (t *Token) factor() *big.Int {
+	var evmFactor int64 = 9
+	if t.EVMFactor != nil {
+		evmFactor = *t.EVMFactor
+	}
+	return new(big.Int).Exp(big.NewInt(10), big.NewInt(evmFactor), nil)
+}
+
+// AtomicToEVM converts from DEX atomic units to EVM units.
+func (t *Token) AtomicToEVM(v uint64) *big.Int {
+	return new(big.Int).Mul(big.NewInt(int64(v)), t.factor())
+}
+
+// EVMToAtomic converts from raw EVM units to DEX atomic units.
+func (t *Token) EVMToAtomic(v *big.Int) uint64 {
+	vDEX := new(big.Int).Div(v, t.factor())
+	if vDEX.IsUint64() {
+		return vDEX.Uint64()
+	}
+	return 0
+}
+
+// NetToken are the addresses associated with the token and it's versioned
```

Comment:
its

Also Joe's comment about the encoding description stands out too.  I guess that was left from when there was more of this stuff in the `dex` package.